mlflow/models/flavor_backend_registry.py

```diff
@@ -38,13 +39,9 @@ def _get_flavor_backend_for_local_model(model=None, build_docker=True, **kwargs)
 def get_flavor_backend(model_uri, **kwargs):
     if model_uri:
         with TempDir() as tmp:
-            if ModelsArtifactRepository.is_models_uri(model_uri):
-                underlying_model_uri = ModelsArtifactRepository.get_underlying_uri(model_uri)
-            else:
-                underlying_model_uri = model_uri
-            local_path = _download_artifact_from_uri(
-                append_to_uri_path(underlying_model_uri, MLMODEL_FILE_NAME), output_path=tmp.path()
-            )
+            root_uri, artifact_path = _get_root_uri_and_artifact_path(model_uri)
+            artifact_repo = get_artifact_repository(root_uri)
+            local_path = artifact_repo.download_artifacts(artifact_path, dst_path=tmp.path())
```

Comment:
Shouldn't we still be downloading the `MLMODEL_FILE_NAME` here?

Reply:
Or maybe `Model.load` is able to figure that out somehow?

Reply:
https://src.dev.databricks.com/mlflow/mlflow@66a2a0c78b6dc5a4cd5518af86b1490459998356/-/blob/mlflow/models/model.py?L593-595

that leads me to believe the work is already being done?  I can put that back in this method if you want tho.

Reply:
Great, no need, was just curious - if tests are passing that means it works :D